src/application/services/useNoteSettings.ts

```diff
@@ -65,14 +65,21 @@ export default function (): UseNoteSettingsComposableState {
   };
 
   /**
-   * Update note settings
+   * Update field isPublic in note settings
    *
    * @param id - Note id
-   * @param data - Note settings data with new values
+   * @param newIsPublicValue - new isPublic
    */
-  const update = async (id: NoteId, data: Partial<NoteSettings>): Promise<void> => {
-    noteSettings.value = await noteSettingsService.patchNoteSettingsByNoteId(id, data);
-  };
+  async function updateIsPublic(id: NoteId, newIsPublicValue: boolean): Promise<void> {
+    const updatedIsPublic = await noteSettingsService.patchNoteSettingsByNoteId(id, { isPublic: newIsPublicValue });
+
+    /**
+     * Check if note setting is not empty
```

Comment:
it still seems like a crutch around TS issue

Reply:
in general, notes are updated without explicitly assigning the result to noteSettings.value, but i think this option is more correct

Reply:
Or I can do like in revokeHash function, smth like:
```
/**
     * Check if note setting is not empty
     */
    if (noteSettings.value) {
      noteSettings.value = { ...noteSettings.value, isPublic };
    }
    ```

Reply:
let's leave it as is